google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryWatchTest.java

```diff
@@ -0,0 +1,550 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.firestore.it;
+
+import static com.google.cloud.firestore.LocalFirestoreHelper.map;
+import static com.google.common.collect.Sets.newHashSet;
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
+import static java.util.Collections.emptySet;
+
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.DocumentChange;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.EventListener;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreException;
+import com.google.cloud.firestore.FirestoreOptions;
+import com.google.cloud.firestore.ListenerRegistration;
+import com.google.cloud.firestore.LocalFirestoreHelper;
+import com.google.cloud.firestore.Query;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.cloud.firestore.it.ITQueryWatchTest.QuerySnapshotEventListener.ListenerAssertions;
+import com.google.common.base.Joiner;
+import com.google.common.base.Joiner.MapJoiner;
+import com.google.common.base.Optional;
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.Range;
+import com.google.common.truth.Truth;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import javax.annotation.Nullable;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+
+public final class ITQueryWatchTest {
+
+  private static Firestore firestore;
+
+  @Rule public TestName testName = new TestName();
+
+  private CollectionReference randomColl;
+
+  @BeforeClass
+  public static void beforeClass() {
+    FirestoreOptions firestoreOptions = FirestoreOptions.newBuilder().build();
+    firestore = firestoreOptions.getService();
+  }
+
+  @Before
+  public void before() {
+    String autoId = LocalFirestoreHelper.autoId();
+    String collPath = String.format("java-%s-%s", testName.getMethodName(), autoId);
+    randomColl = firestore.collection(collPath);
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    firestore.close();
+  }
+
+  /*
+  Attach a listener to a query with empty results.
+  Verify the listener receives an empty event.
+   */
+  @Test
```

Comment:
Unless this is common in GCloud Java, may I recommend changing this to:


```
  /**
   * Attach a listener to a query with empty results.
   * Verify the listener receives an empty event.
   */
```

If you don't think is is JSDoc worthy then I would move it into the method.

Reply:
Will do